test/integration/controller/job/job_controller_test.go

```diff
@@ -275,3 +276,141 @@ var _ = ginkgo.Describe("Job controller for workloads with no queue set", func()
 		}, framework.Timeout, framework.Interval).Should(gomega.Succeed())
 	})
 })
+
+var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
+	const (
+		instanceKey = "cloud.provider.com/instance"
+	)
+
+	var (
+		ns                  *corev1.Namespace
+		onDemandFlavor      *kueue.ResourceFlavor
+		spotTaintedFlavor   *kueue.ResourceFlavor
+		spotUntaintedFlavor *kueue.ResourceFlavor
+		prodClusterQ        *kueue.ClusterQueue
+		devClusterQ         *kueue.ClusterQueue
+		prodQueue           *kueue.Queue
+		devQueue            *kueue.Queue
+	)
+
+	ginkgo.BeforeEach(func() {
+		fwk = &framework.Framework{
```

Comment:
We can place this in the `suite_test.go`, refer to `test/integration/controller/core/suite_test.go`